src/scripts/outlet-registration/index.js

```diff
@@ -0,0 +1,582 @@
+    postInfo.url = postInfo.url.replace("[ORGUNIT]", newOrgUnit.uid);
+    postInfo.url = postInfo.url.replace("[USER]",user.userInfo.id);
+    postInfo.json = true;
+    request.post(postInfo, function(error, response, body){
+        if (error) {console.error("Error adding the orgunit to the user ",error)}
+        console.log(JSON.stringify(body));
+    });
+};
+
+/**
+ * Marks the event as already imported
+ * @param event
+ */
+OutletRegistrator.prototype.markImportedAsTrue = function(event) {
+    var eventToUpdate = this.fillEventToUpdate(event);
+    var putInfo = this.prepareOptions(this.endpoints.EDITEVENTS);
+    putInfo.url = putInfo.url.replace("[EVENT]",event.event);
+    putInfo.body = eventToUpdate;
+    putInfo.json = true;
+    
+    request.put(putInfo, function(error, response, body){
+        if (error) {console.error("Error updating the event ", error)};
+        console.log(JSON.stringify(body));
+    });
+};
+
+/**
+ * Format the event to be updated. It set up 'already imported' to true
+ * @param event
+ */
+OutletRegistrator.prototype.fillEventToUpdate = function(event) {
+    var eventToUpdate = {};
+    var _this = this;
+    
+    eventToUpdate.program = event.program ;
+    eventToUpdate.programStage = event.programStage;
+    eventToUpdate.event = event.event;
+    eventToUpdate.orgUnit = event.orgUnit;
+    eventToUpdate.status = event.status;
+    eventToUpdate.eventDate = event.eventDate.split('T')[0];
+    eventToUpdate.storedBy = event.storedBy;
+    eventToUpdate.coordinate = event.coordinate;
+    eventToUpdate.dataValues = [];
+    event.dataValues.forEach(function(dataValue) {
+        if (dataValue.dataElement != _this.conf.dataElements.alreadyImported)
+            eventToUpdate.dataValues.push({dataElement:dataValue.dataElement, value:dataValue.value});
+    });
+    eventToUpdate.dataValues.push({dataElement:_this.conf.dataElements.alreadyImported, value:"true"});
+    
+    return eventToUpdate;
+};
+
+/**
+ * Check if the org. unit has all its compulsory fields filled
+ * @param orgUnit
+ */
+OutletRegistrator.prototype.ifValidOrgUnit = function(orgUnit) {
+    var validOrgUnit = true;
+    validOrgUnit = validOrgUnit && orgUnit.shortName!=null && orgUnit.shortName.trim()!="";
+    validOrgUnit = validOrgUnit && orgUnit.outletType!=null && orgUnit.outletType.trim()!="";
+    validOrgUnit = validOrgUnit && orgUnit.contactPerson!=null && orgUnit.contactPerson.trim()!="";
+    valirOrgUnit = validOrgUnit && orgUnit.address!=null && orgUnit.address.trim()!="";
+    
+    return validOrgUnit;
+}
+
+/**
+ * Returns the has been already imported or not
+ * @param event The event
+ */
+OutletRegistrator.prototype.isAlreadyImported = function(event) {    
+    var imported = this.findDataValue(event,this.conf.dataElements.alreadyImported);
+    
+    //Not found -> Not imported
+    if(imported === null){
+        return false;
+    }
+    
+    //To avoid parsing issues
+    return JSON.parse(imported);
```

Comment:
This function together with isValidOrgUnit are only called from one place. If you change them into a ternary, maybe it is better if you just remove then and use the ternary instead. I think it will make the code clearer. What do u think?